python/revng/scripting/project.py

```diff
@@ -0,0 +1,67 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+from abc import ABC, abstractmethod
+from copy import deepcopy
+from typing import IO, Tuple
+
+import yaml
+
+from revng.model import Binary
+
+
+class Project(ABC):
```

Comment:
This class is too "thin"
Ideally its descendants would implement some `_fetch_artifact` and `_run_analysis` functions where everything is already figured out and the project would dispatch to the implementations. Right now I see a lot of duplicated functionality because of this.